src/agent/src/rpc.rs

```diff
@@ -1682,7 +1682,10 @@ fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
                 warn!(sl!(), "parse the SigCgt field failed");
                 return false;
             }
-            let sig_cgt_str = mask_vec[1];
+            let mut sig_cgt_str = mask_vec[1];
+            if !sig_cgt_str.chars().nth(0).unwrap().is_numeric() {
```

Comment:
Out of curiosity, can't use simply  trim()` the string?

If that works, that would be cleaner, IMHO.

Note: https://doc.rust-lang.org/std/primitive.str.html#method.trim

Reply:
Nice spot, trim works perfectly, I have updated the code to use this